File: Algoritma/Pengurutan/Selection-Sort.cpp
#include <iostream>

void manual (int *ukuran);
void otomatis (int *ukuran);
void urut_dan_tampilkan (int *ukuran, int array []);

int main ()
{
    int pilihan = 0, ukuran = 0;

    do
    {
        std::cout << "========== SELECTION SORT ==========\n";
        std::cout << "1. Isi array  secara manual\n";
        std::cout << "2. Isi array secara otomatis (acak)\n";
        std::cout << "3. Keluar\n";
        std::cout << "Pilih salah satu: ";
        std::cin >> pilihan;
        std::cin.clear ();
        fflush (stdin);
        
        switch (pilihan)
        {
            case 1: manual (&ukuran);
                    break;
            case 2: otomatis (&ukuran);
                    break;
            case 3: std::cout << "Terima kasih telah menggunakan program ini!\n";
                    break;
            default: std::cout << "Input error\n";
        }
    } while (pilihan != 3);
        
    return 0;
}

void manual (int *ukuran)
{
    std::cout << "Masukkan batas ukuran array: ";
    std::cin >> *ukuran;

    int *array = nullptr;
    array = new int [*ukuran];

    std::cout << "Masukkan nilai tiap elemen array: ";
    for (int i = 0; i < *ukuran; i++)
    {
        std::cin >> array [i];
    }

    urut_dan_tampilkan (ukuran, array);
}

void otomatis (int *ukuran)
{
    int batas = 0;
    srand (time (0));

    std::cout << "Masukkan batas ukuran array: ";
    std::cin >> *ukuran;
    std::cout << "Masukkan batas angka acak: ";
    std::cin >> batas;

    int *array = nullptr;
    array = new int [*ukuran];

    for (int i = 0; i < *ukuran; i++)
    {
       array [i] = rand () % batas + 1;
    }

    std::cout << "Elemen array: ";
    for (int i = 0; i < *ukuran; i++)
    {
       std::cout << array [i] << " ";
    }
    std::cout << "\n";

    urut_dan_tampilkan (ukuran, array);
}

void urut_dan_tampilkan (int *ukuran, int array [])
{
    for (int i = 0; i < *ukuran - 1; i++)
    {
        int minimal = i;
        for (int j = i + 1; j < *ukuran; j++)
        {
            if (array [minimal] > array [j])
            {
                minimal = j;
            }
        }
        int sementara = array [i];
        array [i] = array [minimal];
        array [minimal] = sementara;
    }

    std::cout << "Hasil pengurutan elemen array: ";
    for (int i = 0; i < *ukuran; i++)
    {
        std::cout << array [i] << " ";
    }
    std::cout << "\n";

    delete [] array;
}

File: Algoritma/Pengurutan/Selection Sort/Selection-Sort.py
import random

def main ():
    pilihan = 0
    while pilihan != 3:
        print ("========== INSERTION SORT ==========")
        print ("1. Isi array secara manual")
        print ("2. Isi array secara otomatis (acak)")
        print ("3. Keluar")
        pilihan = int (input ("Pilih salah satu: "))

        if pilihan == 1:
            ukuran = int (input ("Masukkan batas ukuran array: "))
            array = [int (i) for i in input("Masukkan nilai tiap elemen array:\n").split()]
            urut_dan_tampilkan (ukuran, array)
        elif pilihan == 2:
            ukuran = int (input ("Masukkan batas ukuran array: "))
            batas = int (input ("Masukkan batas angka acak: "))
            array = [random.randint (1, batas) for i in range (ukuran)]
            print ("Elemen array:\n")
            print (*array)
            urut_dan_tampilkan (ukuran, array)
        elif pilihan == 3:
            print ("Terima kasih telah menggunakan program ini!")
        else:
            print ("Input error")

def urut_dan_tampilkan (ukuran, array):
    for i in range (ukuran - 1):
        minimal = i
        j = i - 1
        for j in range (i + 1, ukuran):
            if array [minimal] > array [j]:
                minimal = j
        sementara = array [i]
        array [i], array [minimal] = array [minimal], sementara

    print ("Hasil pengurutan elemen array:")
    print (*array)

if __name__ == "__main__":
    main ()